bin/generate.dart

```diff
@@ -348,11 +349,33 @@ class DiagramGenerator {
     }
     if (jobs.isNotEmpty) {
       final ProcessPool pool = ProcessPool();
-      await for (final WorkerJob _ in pool.startWorkers(jobs)) {}
+      await pool.runToCompletion(jobs);
+      _checkJobResults(jobs);
     }
   }
 }
 
+/// Sets the exit code to 1 if at least one of the jobs failed.
+void _checkJobResults(List<WorkerJob> jobs) {
+  if (jobs.any(_hasJobFailed)) {
+    // We're already printing the error message. All that's left is set
+    // the exit code.
+    exitCode = 1;
```

Comment:
can you return 1 here and have the exitCode be set outside this? In general code is easier to reason about if it does not cause changes outside of its own state.

Reply:
Done. Consolidated error reporting into `GeneratorException`.

Reply:
that works. thanks.